### ISSUE TITLE
Unknown errors (ierr) from SETFLUIDSdll and SETREFdll

### ISSUE THREAD

**Issue by schreibfheler** [usnistgov/REFPROP-wrappers#618]
### Description

Calling the SETFLUIDSdll and SETREFdll throw errors that are undocumented in the [REFPROP DLL documentation](https://www.nist.gov/system/files/documents/2018/05/23/refprop10a.pdf) for certain fluids/mixtures. 

#### Examples:
- Diethyl ether;Carbon dioxide ~ 0.9;0.1:           ierr from SETFLUID:    26
- Propane;Methanol ~ 0.9;0.1:                            ierr from SETREF :       -710
- Methane;Ethanol ~ 0.5;0.5:                              ierr from SETREF:        124

While 26 is a valid error code from SETREFdll, it is not listed under SETREFdll.
124 is a valid error code for SATTdll, however, not for SETREF.
-710 is not listed as a valid error code for any dll in the documentation.

``` python
# SETFLUID example code
ierr_SETFLUID = RP.SETFLUIDSdll(hFld=fluid_str)

# SETREF example code
SETREF_output = RP.SETREFdll(hrf="NBP", ixflag=n_comp, x0=z_comp, h0=-1, s0=-1, T0=-1, P0=-1)  
ierr_SETREF = SETREF_output.ierr
```

#### Expected behavior:
The functions should throw errors that can be found in the documentation

#### Actual behavior:
The functions throw errors that cannot be found in the documentation

### Versions

**REFPROP Version:** REFPROP version 10.0
**Operating System and Version:** Windows 10 64-bit
**Access Method:** ctREFPROP 0.10.3

**Comment by ianhbell** [comment]
What are the corresponding `herr` that you get?

**Comment by schreibfheler** [comment]
I don't know why i didn't think of that...

Diethyl ether;Carbon dioxide ~ 0.9;0.1: ierr from SETFLUID: 26

>  SETFLUID doesn't return a herr, so i can't tell.

Propane;Methanol ~ 0.9;0.1: ierr from SETREF : -710

>  [CHECKSTATE warning -710] State appears to be two-phase:  d(ln(P))/d(ln(D)) is positive.

Methane;Ethanol ~ 0.5;0.5: ierr from SETREF: 124

>  [SATT error 124] Iteration for saturation state did not converge for T = 670.429 K.

**Comment by ianhbell** [comment]
For SETFLUIDS, you need to use the ERRMSG function to get the error message. I can replicate the failure:
```python
import os
import ctREFPROP.ctREFPROP as ct
os.environ['RPPREFIX'] = os.getenv('HOME')+'/REFPROP10'

root = os.getenv('RPPREFIX')
RP = ct.REFPROPFunctionLibrary(root)
RP.SETPATHdll(root)
print(RP.RPVersion())

ierr = RP.SETFLUIDSdll("Diethyl ether;Carbon dioxide")
print(ierr)
if ierr != 0:
    print(RP.ERRMSGdll(ierr))
    quit()
    
SETREF_output = RP.SETREFdll(hrf="NBP", ixflag=2, x0=[0.9, 0.1], h0=-1, s0=-1, T0=-1, P0=-1)  
ierr_SETREF = SETREF_output.ierr
print(SETREF_output.herr)
``` 
yielding
```
10.0
0
[SETUP warning 26] Minimum temperature for component 2 is greater than boiling point temperature: Tmin = 216.592 K, Tnbp = 194.686 K; will use default: IIR
```

The last two systems are LLE-containing and REFPROP has a tough time with them. I wouldn't trust those models.

**Comment by schreibfheler** [comment]
How can you tell, the last two systems contain LLE?

For my study, it's not crucial that RP results/properties are "correct" for a given mixture and composition. As long as the mixtures behave like "some" real mixture that might exist in reality, that is ok. E.g. if the saturation curves were leaning more to the right in real life than what RP says, it doesn't matter for me. However if the saturation curves are complete non-sense and no real fluid in the world behaves like this, this would be a problem. Are there any ierr's that might indicate this distinction? -> ierr=-710 -> non-sense properties -> drop mixture from my study.

**Comment by ianhbell** [comment]
I just know, since I have been working with LLE a lot recently. REFPROP explicitly does not consider the possibility of LLE, so it is easy to be misled on that front, as I have learned much to my chagrin.

No, I am afraid not, this is a very tricky question. You need to look at experimental data to know, that is really the only way.

**Comment by schreibfheler** [comment]
I see. Thanks a lot for your help!

Regarding my initially stated issue: is it intended that e.g. -710 cannot be found anywhere in the documentation, and if so, why?

**Comment by ianhbell** [comment]
Because SETREF doesn't expect that sort of an error. My recommendation as always is to ignore reference states entirely. That is especially a mess for mixtures.

**Comment by schreibfheler** [comment]
Okay, thanks again!

**Comment by schreibfheler** [comment]
> Because SETREF doesn't expect that sort of an error. My recommendation as always is to ignore reference states entirely. That is especially a mess for mixtures.

I'd like to clarify what you mean by "ignore reference states entirely." Are you suggesting that it's best not to change the reference state from 'DEF'? In my project, I set it to 'NBP' (Normal Boiling Point), but I've noticed that in 1-2% of cases, the resulting enthalpies are extremely high (hundreds of thousands to millions). This issue seems random, and when I check the mixture in the GUI, the enthalpies appear normal. Could this behavior be related to your recommendation?

**Comment by ianhbell** [comment]
Yes, in my work I _never_ change the reference state. As the reference state is arbitrary, the easiest solution is to never change it, and ignore absolute values of enthalpy and entropy and always calculate them from the model from properties that do not include reference states, like temperature and density.  An enthalpy of a quadrillion could be a totally reasonable value, based on what reference state you select. Hence, best to ignore them. It is an even bigger mess in the mixture case, and there I would even more strongly recommend to not worry about them.
